### PR TITLE
Add "Completed" and "Created"

### PR DESCRIPTION
**Pull request by michaelwehrley**
Add "Completed" and "Created"

* Add "Completed" and "Created" states   
  to a goal's ```action``` method.
* Add specs to cover conditions.

[Finishes: #90287526]

### CODE REVIEW

File: spec/models/social_networking/goals_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+module SocialNetworking
+  RSpec.describe Goal, type: :model do
+    fixtures :all
+
+    let(:goal) { social_networking_goals(:participant1_goal_alpha) }
+
+    describe "#action" do
+      it "returns a 'Created' if the goal was just created" do
+        allow(goal).to receive(:is_completed).and_return(false)
```
Comment (ericcf) [inline spec/models/social_networking/goals_spec.rb]: @michaelwehrley it's usually bad practice to mock/stub the object under test. It couples the spec to the implementation.

File: app/models/social_networking/goal.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+require File.expand_path("../../app/models/social_networking/goal",
+                         SocialNetworking::Engine.called_from)
+
+module SocialNetworking
+  # Extend SocialNetworking::Goal model
+  class Goal
+    def action
+      if is_completed.present?
```
Comment (ericcf) [inline app/models/social_networking/goal.rb]: @michaelwehrley What about simply `is_completed.present? ? "Completed" : "Created"`

Comment (michaelwehrley) [inline app/models/social_networking/goal.rb]: :golf:

File: spec/models/social_networking/goals_spec.rb (hunk 1 above)
Comment (michaelwehrley) [inline spec/models/social_networking/goals_spec.rb]: Thank you! :bow:

**Comment by ericcf** [comment]
@michaelwehrley I always like having these discussions. This little stuff adds up and makes a big difference over time. Thanks!

**Comment by michaelwehrley** [comment]
yep! :hibiscus:

**Comment by michaelwehrley** [comment]
@ericcf is this ready for merging?  Thanks.

**Pull request merged**